cmake/EigenTesting.cmake

```diff
@@ -20,7 +20,8 @@ macro(ei_add_test_internal testname testname_with_suffix)
 
   if(EIGEN_ADD_TEST_FILENAME_EXTENSION STREQUAL cu)
     if(EIGEN_TEST_HIP)
-      hip_add_executable(${targetname} ${filename} HIPCC_OPTIONS "-DEIGEN_USE_HIP")
+      hip_reset_flags()
```

Comment:
what is the purpose of `hip_reset_flags()` here?

Reply:
that was the key to making the `CALL_SUBTEST_{N}` macros work correctly.

`$ARGV2` contains `-DEIGEN_TEST_PART_{N}`, which in turn will enable the corresponding `CALL_SUBTEST_{N}` (and disable the rest). 

Without the call to `hip_reset_flags` , HIPCC_OPTIONS was somehow getting corrupted and the correct `-DEIGEN_TEST_PART_{N}` was not getting specified on the generated command line. This is why I had lumped all subtests into one (i.e. removed the `_{N}` from the `CALL_SUBTEST_{N}`) in the `_hip` versions of the tests. After some digging around in `FindHIP.cmake`, I tried `hip_reset_flags` on a hunch and it seems to have gotten things working.

Reply:
thanks for the clarification